source-repo/poky-ami/build.buildspec.yml

```diff
@@ -63,8 +63,10 @@ phases:
       - find /downloads -atime +30 -type f -delete
       # only delete empty dirs
       - find /downloads -atime +30 -type d -empty -delete
+
 artifacts:
-  discard-paths: true
+  s3-prefix: poky-ami/
```

Comment:
Did you evaluate if there is another possibility to build an S3 path outside of the build spec and pass it into code build? Maybe you can use the objectKey property of the S3DeployActionProps? 

1. What happens if the customer wants to have multiple buckets? We said: default - one bucket and prefix, but if desired, customer could build a bucket per pipeline too. 

2. I have concerns about the implications of having the prefix in the build spec. The build spec is owned by the customer, but the library owns the infrastructure. If we do this in the build spec, we need to document it, and we would need to create a basic proposed build spec template for example to ensure mandatory params are in. What happens if the customer decides to remove the prefix in the build spec? Will data be overwritten?

Reply:
1. he needs to change the buildspec AND app.ts to write to a different bucket
Agree on 2. yes - data will be overwritten.

Still struggling to upload to a s3 bucket subdir - seems that s3-prefix is not the right thing.

Reply:
S3DeployActionProps works